### ISSUE TITLE
Sentry fails with 500 error on anonymous access to publicly shared event

### ISSUE THREAD

**Issue by gthb** [getsentry/sentry#1520]
Accessing e.g. https://sentry.datamarket.com/dm/sentry/group/2614/ (for which I've checked the “Allow anonymous users to view this event” box) fails with a 500 error.

The stack trace is:

```python
Stacktrace (most recent call last):

  File "django/core/handlers/base.py", line 112, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "sentry/web/frontend/groups.py", line 316, in group
    event_id=event_id,
  File "sentry/web/decorators.py", line 218, in wrapped
    return func(request, organization=organization, project=group.project, group=group, *args, **kwargs)
  File "sentry/web/frontend/groups.py", line 425, in group_details
    event=event, is_public=is_public)
  File "sentry/web/frontend/groups.py", line 147, in render_with_group_context
    organization=group.organization,
  File "sentry/auth/access.py", line 75, in from_user
    organization=organization,
  File "django/db/models/manager.py", line 151, in get
    return self.get_queryset().get(*args, **kwargs)
  File "django/db/models/query.py", line 301, in get
    clone = self.filter(*args, **kwargs)
  File "django/db/models/query.py", line 593, in filter
    return self._filter_or_exclude(False, *args, **kwargs)
  File "django/db/models/query.py", line 611, in _filter_or_exclude
    clone.query.add_q(Q(*args, **kwargs))
  File "django/db/models/sql/query.py", line 1204, in add_q
    clause = self._add_q(where_part, used_aliases)
  File "django/db/models/sql/query.py", line 1240, in _add_q
    current_negated=current_negated)
  File "django/db/models/sql/query.py", line 1128, in build_filter
    lookup_type, value)
  File "django/db/models/fields/related.py", line 1107, in get_lookup_constraint
    value[index]), AND)
  File "django/utils/tree.py", line 104, in add
    data = self._prepare_data(data)
  File "django/db/models/sql/where.py", line 79, in _prepare_data
    value = obj.prepare(lookup_type, value)
  File "django/db/models/sql/where.py", line 352, in prepare
    return self.field.get_prep_lookup(lookup_type, value)
  File "django/db/models/fields/__init__.py", line 369, in get_prep_lookup
    return self.get_prep_value(value)
  File "sentry/db/models/fields/bounded.py", line 47, in get_prep_value
    value = int(value)
```

where that `value` is `<SimpleLazyObject: <django.contrib.auth.models.AnonymousUser object at 0xc062dec>>`

This is in version 7.4.3.

**Comment by gthb** [comment]
Fixed locally with:

```diff
diff --git a/src/sentry/auth/access.py b/src/sentry/auth/access.py
index aee6319..371b1f8 100644
--- a/src/sentry/auth/access.py
+++ b/src/sentry/auth/access.py
@@ -66,7 +66,7 @@ def from_user(user, organization):
             sso_is_valid=True,
         )
 
-    if not organization:
+    if user.is_anonymous() or not organization:
         return DEFAULT
 
     try:
```

(Bhe publicly shared event doesn't show anything, though. But that's presumably a separate problem — or a limitation in my workaround, i.e. something other than `DEFAULT` needs to be returned to support public access?)

**Comment by dcramer** [comment]
Looks like we already fixed anonymous access in master. Will double check, but I think things are safe for our 7.5.0 release

**Comment by rogerhu** [comment]
Yeah this was fixed in issue #1485
